runelite-client/src/main/java/net/runelite/client/plugins/combatlevel/CombatLevelOverlay.java

```diff
@@ -0,0 +1,147 @@
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import java.awt.Rectangle;
+import static java.lang.Math.floor;
+
+public class CombatLevelOverlay extends Overlay
+{
+	@Inject
+	private Client client;
+
+	@Inject
+	private CombatLevelConfig config;
+
+	@Inject
+	private TooltipManager tooltipManager;
+
+	@Override
+	public Dimension render(Graphics2D graphics)
+	{
+		if (!config.showLevelsUntil() || client.getLocalPlayer().getCombatLevel() == 126
+			|| client.getVar(VarClientInt.CURRENT_TAB) != 0)
+		{
+			return null;
+		}
+
+		Widget combatLevelWidget = client.getWidget(WidgetInfo.COMBAT_LEVEL);
+		Rectangle combatCanvas = combatLevelWidget.getBounds();
+
+		if (combatCanvas.contains(client.getMouseCanvasPosition().getX(), client.getMouseCanvasPosition().getY()))
+		{
+			tooltipManager.add(new Tooltip(getLevelsUntil()));
+		}
+
+		return null;
+	}
+
+	private String getLevelsUntil()
+	{
+		// put together information
+		int attackLevel = client.getRealSkillLevel(Skill.ATTACK);
+		int strengthLevel = client.getRealSkillLevel(Skill.STRENGTH);
+		int defenceLevel = client.getRealSkillLevel(Skill.DEFENCE);
+		int hitpointsLevel = client.getRealSkillLevel(Skill.HITPOINTS);
+		int magicLevel = client.getRealSkillLevel(Skill.MAGIC);
+		int rangedLevel = client.getRealSkillLevel(Skill.RANGED);
+		int prayerLevel = client.getRealSkillLevel(Skill.PRAYER);
+		double base = 0.25 * (defenceLevel + hitpointsLevel + floor(prayerLevel / 2));
+		double melee = 0.325 * (attackLevel + strengthLevel);
+		double range = 0.325 * (floor(rangedLevel / 2) + rangedLevel);
+		double mage = 0.325 * (floor(magicLevel / 2) + magicLevel);
+		double max = Math.max(melee, Math.max(range, mage));
+
+		// begin calculating levels
+		int next = (int) floor(base + max) + 1;
+		int meleeNeed = calcLevels((base + melee), next, 0.325);
+		int hpdefNeed = calcLevels((base + max), next, 0.25025);
+		int prayNeed = calcLevels((base + max), next, 0.125);
+		int rangeNeed = calcLevelsRM(rangedLevel, next, base);
+		int magicNeed = calcLevelsRM(magicLevel, next, base);
+
+		// create and populate tooltip string
+		StringBuilder sb = new StringBuilder();
+		sb.append("<col=ff981f>Next combat level:</col></br>");
+
+		if ((attackLevel + strengthLevel + meleeNeed) <= 198)
+		{
+			sb.append(meleeNeed).append(" Attack/Strength</br>");
+		}
+		if ((hitpointsLevel + defenceLevel + hpdefNeed) <= 198)
+		{
+			sb.append(hpdefNeed).append(" Defence/Hitpoints</br>");
+		}
+		if ((rangedLevel + rangeNeed) <= 99)
+		{
+			sb.append(rangeNeed).append(" Ranged</br>");
+		}
+		if ((magicLevel + magicNeed) <= 99)
+		{
+			sb.append(magicNeed).append(" Magic</br>");
+		}
+		if ((prayerLevel + prayNeed) <= 99)
+		{
+			sb.append(prayNeed).append(" Prayer");
+		}
+
+		return sb.toString();
+	}
+
+	private static int calcLevels(double start, int end, double multiple)
+	{
+		int need = 0;
+		while (start < end)
+		{
+			start += multiple;
+			need++;
+		}
+		return need;
```

Comment:
is this not `start + ((end - start) / multiple)`?

Reply:
Would this be the return value?

Reply:
It works fine using `Math.ceil((end - start) / multiple)`